thunder/core/rematerialization.py

```diff
@@ -369,23 +366,19 @@ def add_edges(var):
         for var in symbol.flat_proxy_outs:
             add_edges(var)
 
-    g = Graph(
-        n=len(name_to_id),
-        edges=edges,
-        directed=True,
-        edge_attrs={"capacity": capacities},
-    )
-    source = name_to_id["source"]
-    sink = name_to_id["sink"]
+    g = nx.DiGraph()
+    g.add_edges_from(edges)
+
+    _, (set_a, set_b) = nx.minimum_cut(g, "source", "sink")
 
-    id_to_name = dict(map(reversed, name_to_id.items()))
+    cut_edges = []
+    for u in set_a:
+        for v in g.successors(u):
+            if v in set_b:
+                cut_edges.append((u, v))
```

Comment:
Fine by now, but `minimum_cut` is a simple composition of other functions that compute the cut as an intermediary: https://networkx.org/documentation/stable/_modules/networkx/algorithms/flow/maxflow.html#minimum_cut.

Reply:
oh nice

Reply:
turns out `cutset` in there is not the droid we're looking for :)
that is the set of saturated edges that need to be removed from the float graph to obtain the residual flow graph, but we still need to look for unreachable nodes in order to get the actual mincut

Reply:
I am not sure I understand. The flow function is defined on the edges, not on the vertices. So we should be able to extract what we need from this set, modulo removing irrelevant edges from the sink and to the source.

Reply:
yeah, since we have many edges with the  same capacity, we can saturate subgraphs which are paths, and there any single edge can be included in the cut. We do not need all of them, just one will do, might as well be the one closed to the source or the sink. Agree!